R/get_truth.R

```diff
@@ -221,4 +221,76 @@ preprocess_usafacts <- function (save_location="./data-truth/usafacts/"){
       dplyr::rename(value = inc) %>%
       dplyr::select(date, location, value),
     path = paste0(save_location,"truth_usafacts-Incident Cases.csv"))
+}
+
+
+#' Preprocess raw truth data from JHU CSSE into Cumulative/Incident - Deaths/Cases and write to CSVs
+#' 
+#' @param save_location character specifying the location of to save raw truth data.
+#' Default to "./data-truth/"
+#' 
+#' @export
+preprocess_jhu <- function (save_location="./data-truth/"){
+  # Location name
+  location_names <- covidData::fips_codes[, c("location", "location_name")]
+  
+  # National data
+  national_cases_dframes <- covidData::load_jhu_data(spatial_resolution = "national",
+                                                     temporal_resolution = "daily",
+                                                     measure = "cases")
+  national_deaths_dframes <- covidData::load_jhu_data(spatial_resolution = "national",
+                                                      temporal_resolution = "daily",
+                                                      measure = "deaths")
+
+  # State data
+  state_cases_dframes <- covidData::load_jhu_data(spatial_resolution = "state",
+                                                     temporal_resolution = "daily",
+                                                     measure = "cases")
+  state_deaths_dframes <- covidData::load_jhu_data(spatial_resolution = "state",
+                                                      temporal_resolution = "daily",
+                                                      measure = "deaths")
+  
+  # County data
+  county_cases_dframes <- covidData::load_jhu_data(spatial_resolution = "county",
+                                                     temporal_resolution = "daily",
+                                                     measure = "cases")
+  county_deaths_dframes <- covidData::load_jhu_data(spatial_resolution = "county",
+                                                      temporal_resolution = "daily",
+                                                      measure = "deaths")
+  
+  # Combine to get cases data
+  cases_dframes <- dplyr::bind_rows(national_cases_dframes, state_cases_dframes, county_cases_dframes)
+  cases_dframes <- dplyr::left_join(cases_dframes, location_names, by = "location")
+  cases_dframes$inc[cases_dframes$inc < 0] <- 0
+  cases_dframes<-cases_dframes[!(cases_dframes$location==11001),]
```

Comment:
should this filter be against the string `"11001"` rather than the integer `11001`?

Reply:
I don't really know R enough, but I think when R compare a character and a numeric value of the same "value", they return TRUE. I have tried this with `"11001"==11001` and it returns `TRUE`, thus why I was able to use `11001` instead of `"11001"` to filter the dataframe. However, I looked into the data type of the column `location` and it indeed uses character, so I have updated the code to use `"11001"` instead